tests/spec_decode/test_spec_decode_worker.py

```diff
@@ -527,6 +530,165 @@ def test_empty_input_batch(k: int, batch_size: int):
     draft_worker.execute_model.assert_called_once_with(execute_model_req)
     target_worker.execute_model.assert_called_once_with(execute_model_req)
 
+@torch.inference_mode()
+def test_populate_seq_ids_with_bonus_tokens():
```

Comment:
one edge case is where a sequence is skipped but is present in `seq_with_bonus_token_in_last_step`

Reply:
Added this case in Forward pass # 2.